ui/src/app/zones/views/ZonesListTable/ZonesListTable.tsx

```diff
@@ -40,18 +44,26 @@ const ZonesListTable = (): JSX.Element => {
         },
         {
           content: (
-            <Link to={`${machineURLs.machines.index}?zone=%3D${zone.name}`}>
+            <Link to={`${machineURLs.machines.index}${machinesFilter}`}>
               {zone.machines_count}
             </Link>
           ),
           className: "u-align--right",
         },
         {
-          content: zone.devices_count,
+          content: (
+            <LegacyLink route={baseURLs.devices}>
```

Comment:
We also have this `LegacyLink` component specifically for routing to the legacy app. I've noticed that the query params don't actually work in the legacy app, so don't worry about that for now